### PR TITLE
SecretEnricher getSecretsFromXmlConfig should return Collections.emptyList() instead of null 

### LINKED ISSUE #2530: SecretEnricher `getSecretsFromXmlConfig` should return `Collections.emptyList()` instead of `null`
### Component

JKube Kit

### Description
SecretEnricher has this private method which returns `null` that is reported by Sonar as a code smell:
https://github.com/eclipse/jkube/blob/275dc11e9bd0a268aac94f112ac2c443960d957b/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/SecretEnricher.java#L130-L135

We should refactor this method so that it always returns a list. We should return `Collections.emptyList()` instead of `null`. It would be nice to get rid of `if` block using Java'a `Optional.ofNullable`.

### Expected Behavior
`getSecretsFromXmlConfig` is refactored to return a non-null list always

### Acceptance Criteria
- [x] `getSecretsFromXmlConfig` is refactored to return a non `null` value
- [x] `getSecretsFromXmlConfig` uses `Optional.ofNullable` instead of `if` conditional to handle null case
- [x] Remove `null` check from the caller that receives list from `getSecretsFromXmlConfig` 

### Before you start :red_circle:

:point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down::point_down::point_down::point_down::point_down::point_down::point_down:
Make sure you read the [contributing guide](https://www.eclipse.org/jkube/contributing) first. Pay special attention to the ECA agreement section and the requirement to sign-off your commit.

#### Kubernetes

If you don't have a real Kubernetes cluster available (most probably), you can use Minikube or Kind to test with a local cluster.

- Minikube [get started guide](https://minikube.sigs.k8s.io/docs/start/)
- Kind (Requires Docker) [quick start](https://kind.sigs.k8s.io/docs/user/quick-start/)

#### OpenShift

If you don't have a real OpenShift cluster available (most probably), you can use [Red Hat's developer Sandbox for Red Hat OpenShift](https://developers.redhat.com/developer-sandbox). The only requirement is to have a Red Hat account.

Once you have your Sandbox environment, you'll need to download the oc tool from the cluster console.
(Press the `?` icon and from the context menu select `Command line tools`, you'll be redirected to https://$subdomain.openshiftapps.com/command-lines-tools where you'll be able to download the CLI for your platform)

### PR DESCRIPTION
**Pull request by sankritimishra**
## Description
 SecretEnricher getSecretsFromXmlConfig should return Collections.emptyList() instead of null 
Fixes #2530 

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/jkubeio/ci)

Started new GH workflow run for https://github.com/eclipse/jkube/pull/2554 (_2024-01-13T18:09:31Z_)

:gear: [JKube E2E Tests (7514055298)](https://github.com/jkubeio/ci/actions/runs/7514055298) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#2554)](https://github.com/jkubeio/ci/actions/runs/7514055298/job/20456507739)

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/SecretEnricher.java [hunk 1]
```diff
@@ -129,10 +131,7 @@ private void addSecretsFromXmlConfiguration(KubernetesListBuilder builder) {
 
     private List<SecretConfig> getSecretsFromXmlConfig() {
         ResourceConfig resourceConfig = getConfiguration().getResource();
-        if(resourceConfig != null && resourceConfig.getSecrets() != null) {
-            return resourceConfig.getSecrets();
-        }
-        return null;
+        return Optional.ofNullable(resourceConfig).map(resourceConfig1 -> Optional.ofNullable(resourceConfig1.getSecrets()).orElse(Collections.emptyList())).orElse(Collections.emptyList());
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/SecretEnricher.java:134]: minor suggestion regarding formatting
```suggestion
        return Optional.ofNullable(getConfiguration().getResource())
            .map(rc -> Optional.ofNullable(rc.getSecrets())
                .orElse(Collections.emptyList()))
            .orElse(Collections.emptyList());
```

**Review by rohanKanojia — APPROVED** [review]

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/SecretEnricher.java [hunk 2]
```diff
@@ -128,11 +130,10 @@ private void addSecretsFromXmlConfiguration(KubernetesListBuilder builder) {
...
     private List<SecretConfig> getSecretsFromXmlConfig() {
-        ResourceConfig resourceConfig = getConfiguration().getResource();
-        if(resourceConfig != null && resourceConfig.getSecrets() != null) {
-            return resourceConfig.getSecrets();
-        }
-        return null;
+        return Optional.ofNullable(getConfiguration().getResource())
+                .map(rc -> Optional.ofNullable(rc.getSecrets())
+                        .orElse(Collections.emptyList()))
+                .orElse(Collections.emptyList());
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/SecretEnricher.java:136]: @sankritimishra : Sorry for the previous suggestion, [Optional.map](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#map-java.util.function.Function-) takes care of null values by returning an empty optional. You can directly map result without using nested Optional.ofNullable
```suggestion
        return Optional.ofNullable(getConfiguration())
            .map(Configuration::getResource)
            .map(ResourceConfig::getSecrets)
            .orElse(Collections.emptyList());
```

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/SecretEnricher.java [hunk 3]
```diff
@@ -128,11 +130,10 @@ private void addSecretsFromXmlConfiguration(KubernetesListBuilder builder) {
     }
 
     private List<SecretConfig> getSecretsFromXmlConfig() {
-        ResourceConfig resourceConfig = getConfiguration().getResource();
-        if(resourceConfig != null && resourceConfig.getSecrets() != null) {
-            return resourceConfig.getSecrets();
-        }
-        return null;
+        return Optional.ofNullable(getConfiguration())
+            .map(Configuration::getResource)
```
Comment (manusa) [inline jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/SecretEnricher.java:134]: You're missing the required import for `org.eclipse.jkube.kit.enricher.api.model.Configuration`, the project won't compile unless you add it

**Review by manusa — CHANGES_REQUESTED** [review]
Hi, @sankritimishra 
This PR is getting really messy.
I suggest you start for scratch in a new branch (preferably named **different** to `master`).

Keep only these changes:
```java
   private List<SecretConfig> getSecretsFromXmlConfig() {
        return Optional.ofNullable(getConfiguration())
                .map(Configuration::getResource)
                .map(ResourceConfig::getSecrets)
                .orElse(Collections.emptyList());
    }
```

And try to open it on an IDE so that it can automatically fix the imports for you. At the moment I'm worried you're not even trying to compile the changed module in your local machine.

**Pull request closed without merging**